File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries)
{
    vector<int> answer;
    for (int i = 0; i < queries.size(); i++)
    {
        vector<int> temp;   
        for (int j = queries[i][0]; j <= queries[i][1]; j++)
            if (arr[j] > queries[i][2])
                temp.push_back(arr[j]);
        
        sort(temp.begin(), temp.end());
    
        if (!temp.size())
            answer.push_back(-1);
        else
            answer.push_back(temp[0]);
    }
    return answer;
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        
        int idx = 0;
        for (int[] query: queries) {
            int k = query[2];
            int max = Integer.MAX_VALUE;
            for (int i = 0; i < arr.length; i++) {
                if (query[0] <= i && i <= query[1]) {
                    if (arr[i] > k && arr[i] < max) {
                        max = arr[i];
                    }
                }
            }
            answer[idx++] = max == Integer.MAX_VALUE ? -1 : max;
        }
        
        return answer;
    }
}